x-pack/plugins/aiops/common/api/log_categorization/create_category_request.ts

```diff
@@ -106,15 +107,19 @@ export function createCategoryRequest(
   };
 }
 
+type Tokenizer = 'standard' | 'ml_standard';
+
 // This is a copy of the default categorization analyzer but using the 'standard' tokenizer rather than the 'ml_standard' tokenizer.
 // The 'ml_standard' tokenizer splits tokens in a way that was observed to give better categories in testing many years ago, however,
 // the downside of these better categories is then a potential failure to match the original documents when creating a filter for Discover.
 // A future enhancement would be to check which analyzer is specified in the mappings for the source field and to use
 // that instead of unconditionally using 'standard'.
 // However for an initial fix, using the standard analyzer will be more likely to match the results from the majority of searches.
-const categorizationAnalyzer: AggregationsCustomCategorizeTextAnalyzer = {
+const getCategorizationAnalyzer = (
+  tokenizer: Tokenizer = 'standard'
+): AggregationsCustomCategorizeTextAnalyzer => ({
   char_filter: ['first_line_with_letters'],
-  tokenizer: 'standard',
+  tokenizer,
```

Comment:
The `filter` below and the `char_filter` will not be needed is `ml_standard` is passed in, in fact the whole `categorization_analyzer` doesn't need to be set for the default `ml_standard` to be used..

I think rather than passing the tokenizer in, if we have a `useStandardTokenizer` flag as a function param instead.
If true we'll set `categorization_analyzer: categorizationAnalyzer`, if false we don't need to add anything.

Reply:
Thanks, I updated this in [b1434b0](https://github.com/elastic/kibana/pull/176587/commits/b1434b03e17e6e6d9c9f3ee6a22c7a6a39ad1df6), hope I got what you meant 😅 .